consensus/src/block_storage/execution_pool_test.rs

```diff
@@ -0,0 +1,323 @@
+use crate::{
+    block_storage::{BlockReader, BlockStore},
+    test_utils::TreeInserter,
+};
+use aptos_consensus_types::{
+    block::{block_test_utils::certificate_for_genesis, Block},
+    pipelined_block::PipelinedBlock,
+};
+use aptos_crypto::HashValue;
+use aptos_types::{block_info::Round, validator_signer::ValidatorSigner};
+use std::sync::Arc;
+
+/// Helper function to get the [`OrderedBlockWindow`](aptos_consensus_types::pipelined_block::OrderedBlockWindow)
+/// from the `block_store`
+fn get_blocks_from_block_store_and_window(
+    block_store: Arc<BlockStore>,
+    block: &Block,
+    window_size: usize,
+) -> Vec<Block> {
+    let windowed_blocks = block_store
+        .inner
+        .read()
+        .get_block_window(block, window_size);
+    let ordered_block_window = windowed_blocks.unwrap();
+    ordered_block_window.blocks().to_owned()
+}
+
+/// Helper function to create a block tree of size `N` with no forks
+/// ```text
+/// +--------------+       +---------+       +---------+       +---------+       +---------+
+/// | Genesis Block| ----> | Block 1 | ----> | Block 2 | ----> | Block 3 | ----> | Block 4 | --> ...
+/// +--------------+       +---------+       +---------+       +---------+       +---------+
+/// ```
+///
+/// NOTE: `num_blocks` includes the genesis block
+async fn create_block_tree_no_forks<const N: usize>(
+    num_blocks: u64,
+    window_size: usize,
+) -> (TreeInserter, Arc<BlockStore>, [Arc<PipelinedBlock>; N]) {
```

Comment:
Haha this was actually very intentional. I initially used `Vec<Arc<PipelinedBlock>>` but switched to fixed size array. Thanks to array destructuring, fixed size arrays makes it much cleaner to use (no `.unwrap()`) and it will error during compilation if the length of the array changes in any way.